### ISSUE TITLE
BUG - pre-commit breaks (depending on system) with new pre-commit-config.yaml

### ISSUE THREAD

**Issue by TjazVracko** [IRNAS/irnas-zephyr-template#34]
#### Describe the bug

The new pre-commit config specifies `v18.1.7˙ of `clang-format`.

When installing hooks on a new repo, you get this error:
```
pre-commit install --install-hooks
pre-commit installed at .git/hooks/pre-commit
Running in migration mode with existing hooks at .git/hooks/commit-msg.legacy
Use -f to use only pre-commit.
pre-commit installed at .git/hooks/commit-msg
An error has occurred: InvalidManifestError:
==> File /home/tjaz/.cache/pre-commit/repo61yo3iqm/.pre-commit-hooks.yaml
==> At Hook(id='clang-format')
==> At key: types_or
==> At index 10
=====> Type tag 'metal' is not recognized.  Try upgrading identify and pre-commit?
Check the log at /home/tjaz/.cache/pre-commit/pre-commit.log
```

The message is somewhat helpful, it says: `Try upgrading identify and pre-commit?`
Turns out the _"metal"_ type was only introduced in v2.6 of the [identify lib](https://github.com/pre-commit/identify/) (part of the pre-commit project)

The solution is to run:

```
pip install pre-commit --upgrade
pip install identify --upgrade
```

pre-commit itself wrongy specifies `identify>=1.0.0`, so just updating pre-commit does not pull the required version.

------------------------

The [tooling section of the template readme](https://github.com/IRNAS/irnas-zephyr-template?tab=readme-ov-file#tooling) links to the pre-commit install page, which does not mention the identify lib...

I suggest the tooling section directly specifies the 2 commands:
```
pip install pre-commit --upgrade
pip install identify --upgrade
```

@MarkoSagadin Thoughts?

**Comment by MarkoSagadin** [comment]
I agree that this needs to be written down somewhere, it is definitely the most obvious course of action...

I would like to keep the tooling section as minimal as possible, so I would add the above 2 commands to the `irnas-guidelines-repo` and only direct the user in the tooling section to that repo if they also experience additional problems with `pre-commit install` command.

**Comment by TjazVracko** [comment]
The commits above are good. Closing.
